app/views/handlers/relationships/unrelated_question.py

```diff
@@ -1,61 +1,39 @@
 from functools import cached_property
 
-from app.questionnaire.location import Location
-from app.views.handlers.question import Question
+from app.views.handlers.relationships.relationship_question import RelationshipQuestion
 
 
-class UnrelatedQuestion(Question):
+class UnrelatedQuestion(RelationshipQuestion):
     @cached_property
     def list_name(self):
-        parent_block_id = self._schema.parent_id_map[self.block["id"]]
-        return self._schema.get_block(parent_block_id)["for_list"]
+        return self.block["list_summary"]["for_list"]
 
     @cached_property
-    def parent_location(self):
+    def relationships_block(self):
         parent_block_id = self._schema.parent_id_map[self.block["id"]]
-        return Location(
-            section_id=self._current_location.section_id, block_id=parent_block_id
-        )
+        return self._schema.get_block(parent_block_id)
 
     @cached_property
-    def remaining_relationship_list_item_ids(self):
-        list_model = self._questionnaire_store.list_store[self.list_name]
-        start = list_model.index(self._current_location.list_item_id) + 1
-        return list_model[start:]
+    def unrelated_block_id(self):
+        return self.block["id"]
 
     def get_list_summary_context(self):
         return self.list_context(
             self.rendered_block["list_summary"]["summary"],
-            self.rendered_block["list_summary"]["for_list"],
-            for_list_item_ids=self.remaining_relationship_list_item_ids,
+            self.list_name,
+            for_list_item_ids=self.get_remaining_relationships_for_individual(),
         )
 
-    def _get_routing_path(self):
-        return self.router.routing_path(section_id=self.parent_location.section_id)
-
-    def is_location_valid(self):
-        can_access_parent_location = self.router.can_access_location(
-            self.parent_location, self._routing_path
+    def get_remaining_relationships_for_individual(self):
+        list_model = self._questionnaire_store.list_store[self.list_name]
+        previous_location = self.relationship_router.get_previous_location(
+            self.current_location
         )
-
-        if not can_access_parent_location:
-            return False
-
-        if self.current_location.list_name != self.list_name or (
-            self.current_location.list_item_id
-            and not self.router.is_list_item_in_list_store(
-                self.current_location.list_item_id, self.current_location.list_name
-            )
-        ):
-            return False
-
-        return True
-
-    def get_previous_location_url(self):
-        pass
-
-    def get_next_location_url(self):
-        pass  # pragma: no cover
+        previous_item_index = list_model.index(previous_location.to_list_item_id)
+        return list_model[previous_item_index + 1 :]
```

Comment:
The name was a compromise between making it clear and not making it too long. Originally I had `get_remaining_relationship_to_list_item_ids_for_individual`.

I chose `remaining` as I thought of them as the remaining relationships (from the current location) as we are looping over the relationships, rather than any notion of ones that may have been answered.  We could name it something more generic like `get_list_summary_list_item_ids`, but that doesn't give any indication of how we're choosing the list item ids.

Reply:
I don't think there is any clear or nice way we will achieve this so I am happy to keep it as is but maybe we could add a comment about what the method really does?

Reply:
👍  I've added a comment.